crates/diesel_models/src/query/role.rs

```diff
@@ -131,6 +132,97 @@ impl Role {
 
         router_env::logger::debug!(query = %debug_query::<Pg,_>(&query).to_string());
 
+        match generics::db_metrics::track_database_call::<Self, _, _>(
+            query.get_results_async(conn),
+            generics::db_metrics::DatabaseOperation::Filter,
+        )
+        .await
+        {
+            Ok(value) => Ok(value),
+            Err(err) => match err {
+                DieselError::NotFound => {
+                    Err(report!(err)).change_context(errors::DatabaseError::NotFound)
+                }
+                _ => Err(report!(err)).change_context(errors::DatabaseError::Others),
+            },
+        }
+    }
+    pub async fn generic_list_roles_by_entity_type(
+        conn: &PgPooledConn,
+        entity_type: ListRolesByEntityPayload,
+        is_lineage_data_required: bool,
+        limit: Option<u32>,
+    ) -> StorageResult<Vec<Self>> {
+        let mut query = <Self as HasTable>::table().into_boxed();
+
+        match entity_type {
+            ListRolesByEntityPayload::Organization(org_id) => {
+                let entity_in_vec = if is_lineage_data_required {
+                    vec![
+                        EntityType::Organization,
+                        EntityType::Merchant,
+                        EntityType::Profile,
+                    ]
+                } else {
+                    vec![EntityType::Organization]
+                };
+                query = query
+                    .filter(dsl::org_id.eq(org_id))
+                    .filter(
+                        dsl::scope
+                            .eq(RoleScope::Organization)
+                            .or(dsl::scope.eq(RoleScope::Merchant))
+                            .or(dsl::scope.eq(RoleScope::Profile)),
+                    )
+                    .filter(dsl::entity_type.eq_any(entity_in_vec))
```

Comment:
Just for clarity, In case when `is_lineage data_required` `true` case this query will return same result as we will getting without applying this `eq_any` filter?

Reply:
Yes